### PR TITLE
refactor: add new test suite, move test to different test suites, rename test

### PR DESCRIPTION
**Pull request by gaohannk**
Test Audit: https://docs.google.com/document/d/1uPwfzlGtWypXFTimi-hDZhcvAg9H1Xvst7eb257uJqs/edit?resourcekey=0-XiA8Mg7KlCHTLfvXDuRS3w#

Test Suite
- Introduce New Test Suite: preinstalled, license, service
- Remove image_validation Test Suite
- rename image_boot to boot

Test in Test suite
- Move clock_test in service test suite
- Move TestGuestPackages and TestStandardPrograms to preinstalled test suite and name test file as package_test.go
program_test.go
- Move hostname related test to network test suite 
- Move TestHostKeysGeneratedOnce to image_ssh_test.go
- Rename TestBootTime to TestGuestBootTime
- Rename license_test.go to package_license_test.go

No Change:
- disk, security, metadata test suite
- alias_ip_test.go mtu_test.go multinic_minimal_network_test.go in network test suite

### CODE REVIEW

**Comment by gaohannk** [comment]
assign @hopkiw

File: imagetest/test_suites/network/hostname_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+// TestFQDN tests the 'fully qualified domain name', using the logic in the `hostname` utility.
+func TestFQDN(t *testing.T) {
+	image, err := utils.GetMetadata("image")
+	if err != nil {
+		t.Fatalf("couldn't get image from metadata")
+	}
+
+	if strings.Contains(image, "rhel-7-4-sap") {
+		t.Skip("hostname is not working well on RHEL 7-4-Sap")
```
Comment (EricEdens) [inline imagetest/test_suites/network/hostname_test.go:51]: link to bug

**Review by EricEdens — APPROVED** [review]

**Comment by gaohannk** [comment]
Update: Merge preinstalled, service, license test suites into one named configuration. 
Rename metadata to startup_shutdown_script test suites. 
Remove metadata tests.

File: imagetest/test_suites/network/hostname_test.go (hunk 1 above)
Comment (hopkiw) [reply imagetest/test_suites/network/hostname_test.go:51]: i've previously gotten guidance not to include links to internal bugs or other systems in our OSS code

**Pull request closed without merging**